testmodule/providers/jsonrpc/testJsonRpcSubscriptions.nim

```diff
@@ -89,3 +94,58 @@ suite "HTTP polling subscriptions":
     await client.close()
 
   subscriptionTests(subscriptions, client)
+
+suite "HTTP polling subscriptions - filter not found":
+
+  var subscriptions: JsonRpcSubscriptions
+  var client: RpcHttpClient
+  var mockServer: MockRpcHttpServer
+
+  setup:
+    mockServer = MockRpcHttpServer.new()
+    mockServer.start()
+
+    client = newRpcHttpClient()
+    await client.connect("http://" & $mockServer.localAddress()[0])
+
+    subscriptions = JsonRpcSubscriptions.new(client,
+                                             pollingInterval = 100.millis)
+    subscriptions.start()
+
+  teardown:
+    await subscriptions.close()
+    await client.close()
+    await mockServer.stop()
+
+  test "filter not found error recreates filter":
+    let filter = EventFilter(address: Address.example, topics: @[array[32, byte].example])
+    let emptyHandler = proc(log: Log) = discard
+
+    check mockServer.newFilterCounter == 0
+    let jsonId = await subscriptions.subscribeLogs(filter, emptyHandler)
+    let id = string.fromJson(jsonId).tryGet
+    check mockServer.newFilterCounter == 1
+
+    await sleepAsync(300.millis)
```

Comment:
Could maybe make the polling interval 1ms to avoid these long delays.

Reply:
I don't really think that anything <1 second really affects the test runtime that much, especially for such a small project like this, but sure, I will speed it up. But since there are real HTTP requests under the hood happening, I don't think 1ms is good idea.